web/src/components/LoginModal/LoginModal.js

```diff
@@ -0,0 +1,189 @@
+import { useState } from 'react'
+import Dialog from '@material-ui/core/Dialog'
+import Tab from '@material-ui/core/Tab'
+import Tabs from '@material-ui/core/Tabs'
+import InputTextForm from 'src/components/InputTextForm'
+import OutBound from 'src/components/OutBound'
+import { Form, Submit } from '@redwoodjs/forms'
+import { useAuth } from '@redwoodjs/auth'
+import { useFlash } from '@redwoodjs/web'
+import { Link, routes } from '@redwoodjs/router'
+import { subscribe } from 'src/helpers/subscribe'
+
+const LoginModal = ({ open, onClose, shouldStartWithSignup = false }) => {
+  const { logIn, signUp } = useAuth()
+  const { addMessage } = useFlash()
+
+  const [tab, setTab] = useState(shouldStartWithSignup ? 0 : 1)
+  const onTabChange = (_, newValue) => {
+    setTab(newValue)
+    setError('')
+  }
+  const [checkBox, setCheckBox] = useState(true)
+  const [error, setError] = useState('')
+
+  const onSubmitSignUp = async ({ email, password, name, userName }) => {
+    try {
+      setError('')
+      if (checkBox) {
+        subscribe({ email, addMessage })
+      }
+      await signUp({
+        email,
+        password,
+        remember: { full_name: name, userName },
```

Comment:
Any extra data that you want be inside the payload that's passed to the `identity-signup` netlify function needs to be added to this remember property. It will be in the `user_metadata` property in the payload.
I'm opting to continue using `full_name` for the name as this will display nicely as `Name` in the netlify identity dashboard
![image](https://user-images.githubusercontent.com/29681384/101237028-c13b2e00-3729-11eb-81df-23e963257ca2.png)

Why is the data put on a property called remember? I'm not sure but it's pretty confusing since for `login`, `remember` is a boolean value that's mean for persisting the users log in state.

Redwood has in mostly places changed gotrue's ordered arguments to named object arguments.
![image](https://user-images.githubusercontent.com/29681384/101237105-55a59080-372a-11eb-800a-f7bf1a334966.png)
https://github.com/netlify/gotrue-js/blob/d1565a7d0576ff613b1c37c46a42d1fcbd720c7c/index.d.ts
Which I'd normally like, but with the name change it becomes confusing figuring how things differ from gotrue's docs, this might be side-effect of integrating with multiple auth providers and trying to unify the API for all of them, which I can't fault. I just got caught on this for a bit.